### ISSUE TITLE
Checkstyle xml is invalid

### ISSUE THREAD

**Issue by OrangeDog** [angular-eslint/angular-eslint#303]
### Description
As there's no way to specify output to a file the only option is to redirect stdout. However, if there are multiple files with issues the checkstyle formatter generates multiple xml documents.

### Reproduction
From a new project, add two files with lint issues and run 
```sh
ng lint --format=checkstyle >checkstyle-result.xml
```

### Exception or error
The output is invalid, being a concatenation of multiple documents:
```xml
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="/opt/proj/src/app/app-routing.module.ts"><error line="8" column="35" severity="warning" message="Expected object keys to be in ascending order. &apos;pathMatch&apos; should be before &apos;redirectTo&apos;. (sort-keys)" source="eslint.rules.sort-keys" /><error line="9" column="18" severity="warning" message="Expected object keys to be in ascending order. &apos;component&apos; should be before &apos;path&apos;. (sort-keys)" source="eslint.rules.sort-keys" /><error line="10" column="30" severity="warning" message="Expected object keys to be in ascending order. &apos;component&apos; should be before &apos;path&apos;. (sort-keys)" source="eslint.rules.sort-keys" /><error line="11" column="29" severity="warning" message="Expected object keys to be in ascending order. &apos;component&apos; should be before &apos;path&apos;. (sort-keys)" source="eslint.rules.sort-keys" /></file></checkstyle>
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="/opt/proj/src/app/graphics/graphics.component.ts"><error line="50" column="37" severity="warning" message="Expected object keys to be in ascending order. &apos;start&apos; should be before &apos;types&apos;. (sort-keys)" source="eslint.rules.sort-keys" /><error line="50" column="50" severity="warning" message="Expected object keys to be in ascending order. &apos;end&apos; should be before &apos;start&apos;. (sort-keys)" source="eslint.rules.sort-keys" /><error line="113" column="49" severity="warning" message="Expected object keys to be in ascending order. &apos;lat&apos; should be before &apos;lng&apos;. (sort-keys)" source="eslint.rules.sort-keys" /></file></checkstyle>
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="/opt/proj/src/app/graphics/plot-select.component.spec.ts"><error line="25" column="7" severity="warning" message="Expected object keys to be in ascending order. &apos;endTime&apos; should be before &apos;startTime&apos;. (sort-keys)" source="eslint.rules.sort-keys" /><error line="40" column="7" severity="warning" message="Expected object keys to be in ascending order. &apos;endTime&apos; should be before &apos;startTime&apos;. (sort-keys)" source="eslint.rules.sort-keys" /></file></checkstyle>
```

### Environment
```plain
Angular CLI: 11.0.5
Node: 14.15.0
OS: win32 x64

Angular: 11.0.5
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1100.5
@angular-devkit/build-angular   0.1100.5
@angular-devkit/core            11.0.5
@angular-eslint/builder         1.0.0
@angular-devkit/schematics      11.0.5
@angular/cdk                    11.0.3
@angular/material               11.0.3
@schematics/angular             11.0.5
@schematics/update              0.1100.5
rxjs                            6.6.3
typescript                      4.0.5
```

### Workaround
It's possible to pipe to `split` to generate multiple valid files. A single valid file to start with would be preferable.
```sh
ng lint --format=checkstyle | split -l 1 - checkstyle-result.xml.
```

### See also
https://github.com/angular/angular-cli/issues/14659
https://github.com/angular/angular-cli/issues/4959

**Comment by AlfaCoder** [comment]
+1

**Comment by jakubsturc** [comment]
I believe all formatters are affected by this issue. For example, you can try `ng lint --format=table`.

I believe the root cause hides in [./packages/builder/src/index.ts /](https://github.com/angular-eslint/angular-eslint/blob/master/packages/builder/src/index.ts) where we are calling `formatter.format([result])` for each result individually instead of calling it once for whole `lintResults`.

**Comment by OrangeDog** [comment]
> I believe all formatters are affected by this issue.

That was the conclusion here, but it got closed anyway: https://github.com/angular/angular-cli/issues/14659
I tried opening a new one but it also got closed and I was directed here.

**Comment by JamesHenry** [comment]
The current behaviour was informed by the Angular CLI's TSLint builder way of doing things.

I agree with changing things so that format() is called once on the results list so this will help with the issue. I have a PR in the works for that change.

It's worth noting though, that if you have a workspace with multiple projects in it, you still won't be able to combine all of their results into a single output because of the fact that linting via `ng lint` takes place at a project, not a workspace, level.

If you want to have a single output for everything in a workspace, you would need to refactor your approach to linting to move away from the being executed by `ng` and towards invoking `eslint` directly via its inbuilt CLI.

**Comment by JamesHenry** [comment]
The format at a project level is addressed in v2, which is in PR form here: https://github.com/angular-eslint/angular-eslint/pull/358

I am publishing prereleases from it for those who are feeling brave - no migration advice will be provided until it is complete but my plan is for v2 stable to go out sometime this week.

The point around combining all projects into a single report still applies, because that is fundamental to how the Angular CLI is intended to work.

**Comment by JamesHenry** [comment]
Resolved in the newly published `v2.0.0`, see releases or changelog for more information.
